src/chains/evm/spoke/modules/folks-evm-gmp.ts

```diff
@@ -70,6 +78,62 @@ export const prepare = {
       bridgeRouterAddress: spokeChain.bridgeRouterAddress,
     };
   },
+
+  async resendWormholeMessage(
+    provider: Client,
+    sender: EvmAddress,
+    sourceFolksChainId: FolksChainId,
+    emitterAddress: GenericAddress,
+    sequence: bigint,
+    targetFolksChainId: FolksChainId,
+    receiverValue: bigint,
+    receiverGasLimit: bigint,
+    transactionOptions: EstimateGasParameters = {
+      account: sender,
+    },
+  ): Promise<PrepareResendWormholeMessageCall> {
+    const { wormholeChainId: sourceWormholeChainId, wormholeRelayer: wormholeRelayerAddress } =
+      getWormholeData(sourceFolksChainId);
+    const wormholeRelayer = getWormholeRelayerContract(provider, wormholeRelayerAddress);
+    const { wormholeChainId: targetWormholeChainId } = getWormholeData(targetFolksChainId);
+
+    // get delivery provider address
+    const evmDeliveryProviderAddress = (await wormholeRelayer.read.getDefaultDeliveryProvider()) as EvmAddress;
+    const genericDeliveryProviderAddress = convertToGenericAddress(evmDeliveryProviderAddress, ChainType.EVM);
+
+    // get quote for delivery
+    const [value] = (await wormholeRelayer.read.quoteEVMDeliveryPrice([
+      targetWormholeChainId,
+      receiverValue,
+      receiverGasLimit,
+      evmDeliveryProviderAddress,
+    ])) as [bigint, bigint];
+
+    // estimate gas limit
+    const vaaKey = {
+      chainId: sourceWormholeChainId,
+      emitterAddress,
```

Comment:
Does Wormhole use the same format as our `GenericAddress` on the relayer?

Reply:
yes